### PR TITLE
feat(generator): add server extensions for router, middleware, binding, and stubs

### PR DESCRIPTION
**Pull request by erraggy**
## Summary

Add comprehensive server-side code generation extensions that integrate with `httpvalidator` for runtime validation, parameter binding, routing, and testing.

### New CLI Flags (require `--server`)

| Flag | Generated File | Description |
|------|----------------|-------------|
| `--server-responses` | `server_responses.go` | Typed response writers and helpers |
| `--server-binder` | `server_binder.go` | Type-safe parameter binding from HTTP requests |
| `--server-middleware` | `server_middleware.go` | Validation middleware using httpvalidator |
| `--server-router` | `server_router.go` | HTTP router with path matching (stdlib or chi) |
| `--server-stubs` | `server_stubs.go` | Configurable stub implementations for testing |
| `--server-all` | All above | Enable all server extensions |

### Example Usage

```bash
# Generate complete server with all extensions
oastools generate --server --server-all -o ./server -p api openapi.yaml

# Generate server with specific extensions
oastools generate --server --server-router=stdlib --server-middleware -o ./server -p api openapi.yaml
```

### Generated Code Features

- **Response Helpers**: Per-operation response types with `Status200()`, `StatusDefault()` methods
- **Request Binder**: Type-safe parameter extraction with validation error details
- **Validation Middleware**: Request/response validation with configurable error handling
- **HTTP Router**: Dispatches requests to `ServerInterface` implementation
- **Stub Server**: Configurable mock implementation for testing with functional options

## Test plan

- [x] All 4448 existing tests pass
- [x] 9 new tests for server extensions
- [x] 87.5% generator coverage
- [x] Tests verify each extension generates expected content
- [x] Edge cases: empty paths, pointer types, combined options

## Changes

- 15 files changed, 2339 insertions
- 5 new templates in `generator/templates/server/`
- Documentation updated (README, deep_dive.md, cli-reference.md)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: generator/templates/server/router.go.tmpl [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+{{range .PathParams}}
+	// Extract path parameter: {{.Name}}
+	srvReq.{{.FieldName}} = PathParam(req, "{{.Name}}")
+{{end}}
+
+	// Call the server implementation
+	resp, err := r.server.{{.MethodName}}(ctx, srvReq)
+	if err != nil {
+		WriteError(w, http.StatusInternalServerError, err.Error())
+		return
```
[the remark this answers is not in the document]
Comment (erraggy) [reply generator/templates/server/router.go.tmpl]: addressed with 40cd2d7
